app/src/main/java/org/onionshare/android/tor/TorManager.kt

```diff
@@ -298,66 +313,56 @@ class TorManager @Inject constructor(
     private suspend fun checkStartupProgress() {
         LOG.info("Starting check job")
 
-        var usedBuiltInBridges = false
-        runWhenMoatTimeIsUp {
-            val bridges = try {
-                getBridgesFromMoat()
+        runWhenStartingTimedOut {
+            LOG.info("Getting bridges from Moat...")
+            try {
+                getBridgesFromMoat().takeIf { it.isNotEmpty() }
             } catch (e: IOException) {
                 LOG.warn("Error getting bridges from moat: ", e)
                 null
```

Comment:
When this happens, the `let` block below isn't executed, so we return from this `runWhenStartingTimedOut` block without changing the config, right? And then we pass the block at line 329 to `runWhenStartingTimedOut`, which waits for another timeout before using built-in bridges. So it seems like we're waiting unnecessarily long before using built-in bridges if Moat fails?

Reply:
Or you could say, if moat doesn't give us bridges, we are waiting extra hard and keep the code simple ;)

Anyway, I pushed a new commit that makes this a bit more complicated to avoid extra hard waiting.

Reply:
I've added a commit that hopefully simplifies this and solves an issue with timeouts that I didn't spot earlier.

Reply:
Looks great, thanks a lot for this improvement!